cli/cmd/admin/start.go

```diff
@@ -377,3 +391,18 @@ func StartCmd(ch *cmdutil.Helper) *cobra.Command {
 	}
 	return startCmd
 }
+
+func loadDatabaseEncryptionKeyring(keyRingConfig string) ([]*database.EncryptionKey, error) {
+	if keyRingConfig == "" {
+		return nil, nil
+	}
+
+	var keyRing []*database.EncryptionKey
+
+	err := json.Unmarshal([]byte(keyRingConfig), &keyRing)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse encryption keyring: %w", err)
+	}
+
+	return keyRing, nil
+}
```

Comment:
Maybe we should push this parsing into the `database` package? Could have `database.NewRandomKeyring() string` and `database.ParseKeyring(keyring string) []*database.EncryptionKey`? Maybe even pass `keystring string` to `database.Open` and let it do the parsing there.

Reply:
This could also be used to add a script in `scripts/generate_db_keyring` for generating a safe keyring to set in Vault (similar to `scripts/generate_jwks`).

Reply:
good suggestions, added